Gordon360/Services/ProfileService.cs

```diff
@@ -274,6 +276,111 @@ public async Task UpdateCustomProfileAsync(string username, string type, CUSTOM_
+    public PublicFacultyStaffProfileViewModel ToPublicFacultyStaffProfileViewModel
+        (string username, string currentUserType, FacultyStaffProfileViewModel fac)
+    {
+        PublicFacultyStaffProfileViewModel publicFac = (PublicFacultyStaffProfileViewModel)fac;
+        var account = accountService.GetAccountByUsername(username);
+
+        // select all privacy settings
+        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+        // get type of viewmodel for reflection property setting
+        Type fs_vm = new PublicFacultyStaffProfileViewModel().GetType();
+
+        foreach (UserPrivacy_Settings row in privacy)
+        {
+            if (row.Visibility == "Private" || (row.Visibility == "FacStaff" && currentUserType != "fac"))
+            {
+                fs_vm.GetProperty(row.Field).SetValue(publicFac, "Private as requested.");
+            }
+        }
+
+        return publicFac;
+    }
+
+    /// <summary>
+    /// convert original student profile to public student profile based on individual privacy settings
+    /// </summary>
+    /// <param name="username">username of the student being searched</param>
+    /// <param name="currentUserType">personnel type of the logged-in user</param>
+    /// <param name="stu">original profile of the student being searched</param>
+    /// <returns>public profile of the student based on individual privacy settings</returns>
+    public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
+        (string username, string currentUserType, StudentProfileViewModel stu)
+    {
+        PublicStudentProfileViewModel publicStu = (PublicStudentProfileViewModel)stu;
+        var account = accountService.GetAccountByUsername(username);
+
+        // select all privacy settings
+        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+        // get type of viewmodel for reflection property setting
+        Type s_vm = new PublicStudentProfileViewModel().GetType();
+
+        foreach (UserPrivacy_Settings row in privacy)
+        {
+            if (row.Visibility == "Private" || (row.Visibility == "FacStaff" && currentUserType != "fac"))
+            {
+                s_vm.GetProperty(row.Field).SetValue(publicStu, "Private as requested.");
+            }
+        }
+
+        return publicStu;
+    }
+
+    /// <summary>
+    /// get privacy setting for particular user
+    /// </summary>
+    /// <param name="username">AD username</param>
+    /// <returns>all privacy setting of the particular user</returns>
+    public IEnumerable<UserPrivacyViewModel> GetPrivacySettingAsync(string username)
+    {
+        var account = accountService.GetAccountByUsername(username);
+
+        // select all privacy settings
+        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID).Select(up_s => (UserPrivacyViewModel)up_s);
```

Comment:
```suggestion
        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID).Select<UserPrivacy_Settings, UserPrivacyViewModel>(u => u);
```
I believe this can be cleaned up a bit to something like the following. It is clearer what is happening here IMO.

Make sure to test my syntax as I am not 100% sure.